streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -83,9 +84,27 @@ public void register(final StateRestorer restorer) {
 
         final Set<TopicPartition> restoringPartitions = new HashSet<>(needsRestoring.keySet());
         try {
-            final ConsumerRecords<byte[], byte[]> allRecords = poll(restoreConsumer, 10);
-            for (final TopicPartition partition : restoringPartitions) {
-                restorePartition(allRecords, partition, active.restoringTaskFor(partition));
+            updatedEndOffsets = !needsRestoring.isEmpty() ? 
+                restoreConsumer.endOffsets(restoringPartitions) : updatedEndOffsets;
+            while (!needsRestoring.isEmpty()) {
```

Comment:
We don't want to do it this way, because, we want restore and calling `mainCosumer.poll()` to interleave -- otherwise, we might drop out of the consumer group as restore is expected to take longer than `max.poll.intervall.ms`. Hence, within this method, we should only do a single `poll(restoreCosumer, 10)` an return afterwards -- the main loop will make sure that this method will be called again to resume the restore.